registry/github_registry.go

```diff
@@ -38,6 +39,7 @@ type githubRegistry struct {
 	format         common.RegistryFormat
 	credentialName string
 	service        GithubRepositoryService
+	httpClient     *http.Client
```

Comment:
GithubRepositoryService comes from [github.Client](https://github.com/google/go-github/blob/master/github/github.go#L83). So, instead of passing around the service, we can pass around the github.Client, and then pull the service when we need it. http.Client is also available from github.Client, so no increase in the argument count.